pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper.go

```diff
@@ -77,9 +77,9 @@ func getBranches(
 ) error {
 	defer waitGroup.Done()
 	branches, err := getBranchNames(context.Background(), graphClient, projectPath)
-
-	ch <- branchData{ProjectPath: projectPath, BranchNames: branches.Project.Repository.BranchNames}
-
+	if err != nil {
+		ch <- branchData{ProjectPath: projectPath, BranchNames: branches.Project.Repository.BranchNames}
+	}
 	return err
```

Comment:
This is a way to do it for sure. my question is should this function return an error at all?